src/core/public/application/application_service.tsx

```diff
@@ -294,14 +306,13 @@ export class ApplicationService {
         const relUrl = http.basePath.prepend(getAppUrl(availableMounters, appId, path));
         return absolute ? relativeToAbsolute(relUrl) : relUrl;
       },
-      navigateToApp: async (appId, { path, state }: { path?: string; state?: any } = {}) => {
-        if (await this.shouldNavigate(overlays)) {
-          if (path === undefined) {
-            path = applications$.value.get(appId)?.defaultPath;
-          }
-          this.appLeaveHandlers.delete(this.currentAppId$.value!);
-          this.navigate!(getAppUrl(availableMounters, appId, path), state);
-          this.currentAppId$.next(appId);
+      navigateToApp,
+      navigateToUrl: async url => {
```

Comment:
not 100% sure we should, but we could make a polymorphic function
```js
navigateTo(app: App): void
navigateTo(url: string): void
```

Reply:
Not sure either. Upside from the polymorphic approach would mostly be reduced API exposure and 'smarter' API. Downside is a (little) less explicit API (`navigateTo` vs `navigateToXXX`), and a slightly more complex implementation as the signature got different number of parameters.

```ts
navigateTo(appId: string, options?: { path?: string; state?: any }): void
navigateTo(url: string): void
```

Probably outside of the scope of this PR as it gonna impact all current calls from plugin code, but shall I open a follow up issue to discuss this possibility?